### PR TITLE
Authentication(logIn/logOut)

### PR DESCRIPTION
**Pull request by PraepechPrAe**
- check username
- compare based password
- set cookie
- remove cookie

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { UsersService } from '../users/users.service';
+import { JwtService } from '@nestjs/jwt';
+import * as argon2 from "argon2";
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService,
+              private jwtService: JwtService) {}
+
+  async signIn(username: string, pass: string): Promise<any> {
```
Comment (ishiasaka) [inline src/auth/auth.service.ts]: Why `Promise<any>`?

File: src/main.ts [hunk 2]
```diff
@@ -21,6 +22,10 @@ async function bootstrap() {
   });
   app.useGlobalPipes(new ValidationPipe({ transform: true }));
 
+  await app.register(fastifyCookie, {
+    secret: 'my-secret', // for cookies signature
```
Comment (ishiasaka) [inline src/main.ts:26]: please use value `process.env.COOKIE_SECRET ?? ''`

File: src/users/users.service.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      userId: 2,
+      username: 'maria',
+      password: 'guess',
+    },
+  ];
+
+  async findOne(username: string): Promise<User | undefined> {
+    return this.users.find(user => user.username === username);
+  }
+}
```
Comment (ishiasaka) [inline src/users/users.service.ts]: Connect to db

File: src/auth/auth.controller.spec.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      controllers: [AuthController],
+    }).compile();
+
+    controller = module.get<AuthController>(AuthController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+});
```
Comment (ishiasaka) [inline src/auth/auth.controller.spec.ts]: If you don't want to unit test please remove it

File: src/auth/auth.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      providers: [AuthService],
+    }).compile();
+
+    service = module.get<AuthService>(AuthService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+});
```
Comment (ishiasaka) [inline src/auth/auth.service.spec.ts]: If you don't want to unit test please remove it

**Review by ishiasaka — CHANGES_REQUESTED** [review]

File: src/auth/auth.service.spec.ts (hunk 5 above)
Comment (PraepechPrAe) [reply src/auth/auth.service.spec.ts]: Ok, deleted

File: src/auth/auth.controller.spec.ts (hunk 4 above)
Comment (PraepechPrAe) [reply src/auth/auth.controller.spec.ts]: Ok, deleted

**Review by PraepechPrAe — COMMENTED** [review]
Beautiful

**Review by ishiasaka — DISMISSED** [review]

**Review by ishiasaka — APPROVED** [review]

**Pull request merged**